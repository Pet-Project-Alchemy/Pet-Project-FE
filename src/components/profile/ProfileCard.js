import React from 'react';
import PropTypes from 'prop-types';
import './ProfileCard.scss';
import defaultUserImg from '../assets/profile.png';
import defaultDogImage from '../assets/dog.png';


const ProfileCard = ({
  firstName,
  dogName,
  userImg,
  dogImg,
  timeNeeded,
  timeAvailable,
  bio,
  dogBio
}) => {
  return (
    <div className='ProfileCard'>
      <div className='ProfileCard__side ProfileCard__side--front'>
        <div className='ProfileCard__picture ProfileCard__picture--1'>
          <div className='ProfileCard__picture ProfileCard__img'>
            <img className='userImg' src={userImg ? userImg : defaultUserImg} />
            <figcaption>
              <span className='ProfileCard__heading-span ProfileCard__heading-span--1'>
                {firstName}
              </span>
            </figcaption>
            <img className='dogImg' src={dogImg ? dogImg : defaultDogImage} />
            <figcaption>
              <span className='ProfileCard__heading-span ProfileCard__heading-span--2'>
                {dogName}&nbsp;
              </span>
            </figcaption>
          </div>
        </div>
        <div className='ProfileCard__details'>
          <ul>
            <li>I can help {timeAvailable}</li>
            <li>I need help {timeNeeded}</li>
          </ul>
        </div>
      </div>
      <div className='ProfileCard__side ProfileCard__side--back ProfileCard__side--back-1'>
        <div className='ProfileCard__cta'>
          <div className='ProfileCard__bio-box'>
            <p className='ProfileCard__bio-text'>{bio}</p>
            <p className='ProfileCard__bio-text'>{dogBio}</p>
          </div>
        </div>
      </div>
    </div>
  );
};

ProfileCard.propTypes = {
  firstName: PropTypes.string.isRequired,
  dogName: PropTypes.string.isRequired,
  userImg: PropTypes.string,
  dogImg: PropTypes.string,
  profileImg: PropTypes.string,
  timeNeeded: PropTypes.string.isRequired,
  timeAvailable: PropTypes.string.isRequired,
  zipcode: PropTypes.string.isRequired,
  bio: PropTypes.string.isRequired,
  dogBio: PropTypes.string.isRequired
};
export default ProfileCard;
