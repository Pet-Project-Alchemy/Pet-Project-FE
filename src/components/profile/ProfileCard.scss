@import '../abstract/base.scss';
@import '../abstract/variables.scss';
@import '../abstract/mixins.scss';

.ProfileCard {
  display: grid;
  perspective: 150rem;
  -moz-perspective: 150rem;
  position: relative;
  height: 43rem;
  width: 28rem;
  
  &__side {
    height: 43rem;
    font-size: 2rem;
    transition: all 0.8s ease;
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    backface-visibility: hidden;
    border-radius: 3px;
    overflow: hidden;
    box-shadow: 0 1.5rem 4rem rgba($color-black, 0.15);
    
    &--front {
      background-color: $color-primary;
    }
    @media all and (max-width: 699px) and (min-width: 520px), (min-width: 900px) {
    &--back {
      transform: rotateY(180deg);
      &-1 {
        background: linear-gradient(
            170deg,
            rgba($color-darkgrey, 0.6) 0%,
            rgba($color-darkgrey, 0.4) 100%,
            transparent 80%
          ),
          url(../assets/dogPhoto/sidePaw.png) left/cover;
        padding: 2rem 0rem;
        border-radius: 3px;
      }
    }
  }
  }
 
    
  @media all and (max-width: 699px) and (min-width: 520px), (min-width: 900px) {
    &:hover &__side--front {
      transform: rotateY(-180deg);
    }
  }
  
@media all and (max-width: 699px) and (min-width: 520px), (min-width: 900px) {
  &:hover &__side--back {
    transform: rotateY(0);
  }
}

  &__picture {
    height: 27rem;
    clip-path: polygon(0 0, 100% 0, 100% 80%, 0 100%);
    background-clip: polygon(0 0, 100% 0, 100% 90%, 0 100%);
    border-top-left-radius: 3px;
    border-top-right-radius: 3px;
    &--1 {
      background: linear-gradient(
          to bottom right,
          $color-primary,
          $color-black-clear
        ),
        url(../assets/dogPhoto/dog.png) center/cover no-repeat;
    }
  }
  &__heading-span {
    padding: 1rem;
    text-transform: capitalize;
    color: $color-black;
  }
  &__heading-span--2 {
    text-transform: capitalize;
  }
  &__details {
    padding: 2rem;
    @include respond(tab-port) {
      padding: 0.2rem;
    }
    ul {
      list-style: none;
      width: 80%;
      margin: 1rem auto;
      li {
        text-align: center;
        font-size: 1.5rem;
        padding: 0.4rem;
        color: $color-black;
        &:not(:last-child) {
          border-bottom: 1px solid $color-grey-med;
        }
      }
    }
  }
  
  &__bio-box {
    text-align: center;
    color: $color-black;
    margin: 2rem;
  }
  &__bio-text {
    font-size: 1.4rem;
    padding: 0.5rem;
    color: black;
    @include respond(tab-port) {
      display: none;
    }
  }
}
.userImg,
.dogImg {
  filter: grayscale(1);
  opacity: 0.8;
  top: 2.5rem;
  left: 2.5rem;
  margin: 1rem 0.5rem 0 1rem;
  height: 8rem;
  width: 8rem;
  border-radius: 50%;
}
.mediaText {
  font-size: 1.4rem;
  color:white;
  text-align: center;
  margin: 0 auto;
  overflow: hidden;
}

.card-wrapper{
  display: flex;
  justify-content: center;
  
}

