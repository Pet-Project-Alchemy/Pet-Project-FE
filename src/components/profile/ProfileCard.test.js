import React from 'react';
import { shallow } from 'enzyme';
import ProfileCard from './ProfileCard';

describe('ProfileCard component', () => {
  it('renders ProfileCard', () => {
    const wrapper = shallow(<ProfileCard />);
    expect(wrapper).toMatchSnapshot();
  });
});
