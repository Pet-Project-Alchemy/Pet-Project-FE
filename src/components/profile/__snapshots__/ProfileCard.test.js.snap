// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileCard component renders ProfileCard 1`] = `
<div
  className="card"
>
  <div
    className="card__side card__side--front"
  >
    <div
      className="card__picture card__picture--1"
    >
      <div
        className="card__picture card__img"
      >
        <img
          className="userImg"
          src="/path/image"
        />
        <figcaption>
          <span
            className="card__heading-span card__heading-span--1"
          >
             &
          </span>
        </figcaption>
        <section
          className="dogImg"
          style={
            Object {
              "backgroundImage": "url(undefined)",
            }
          }
        />
        <figcaption>
          <span
            className="card__heading-span card__heading-span--2"
          >
             
          </span>
        </figcaption>
      </div>
    </div>
    <div
      className="card__details"
    >
      <ul>
        <li>
          I can help 
        </li>
        <li>
          I need help 
        </li>
      </ul>
    </div>
  </div>
  <div
    className="card__side card__side--back card__side--back-1"
  >
    <div
      className="card__cta"
    >
      <div
        className="card__bio-box"
      >
        <p
          className="card__bio-text"
        />
        <p
          className="card__bio-text"
        />
      </div>
    </div>
  </div>
</div>
`;
